src/Catty/ViewController/Help/HelpWebViewController.swift

```diff
@@ -142,111 +162,128 @@ class HelpWebViewController: UIViewController, UIWebViewDelegate, UIScrollViewDe
+
+        let request = navigationAction.request
         if !((request.url?.absoluteString ?? "").contains(NetworkDefines.downloadUrl)) {
-            return true
+            decisionHandler(.allow)
+            return
         }
+
         let urlWithoutParams: String? = request.url?.absoluteString
             .components(separatedBy: CharacterSet(charactersIn: "?"))[0]
+
         // extract project ID from URL => example: https://pocketcode.org/download/959.catrobat
         var urlParts = urlWithoutParams?.components(separatedBy: "/")
         if urlParts?.isEmpty ?? true {
             Util.alert(withText: kLocalizedInvalidURLGiven)
-            return false
+            decisionHandler(.cancel)
+            return
         }
+
         // get last part of url and split by using separator "." => 959.catrobat
         urlParts = urlParts?.last?.components(separatedBy: ".")
         if urlParts?.count != 2 {
             Util.alert(withText: kLocalizedInvalidURLGiven)
-            return false
+            decisionHandler(.cancel)
+            return
         }
+
         // check projectID
         let projectID = urlParts?.first ?? ""
         if projectID.isEmpty {
             Util.alert(withText: kLocalizedInvalidURLGiven)
-            return false
+            decisionHandler(.cancel)
+            return
         }
+
         // get url parameters
         guard let urlComp = URLComponents(string: (request.url?.absoluteString)!) else {
             Util.alert(withText: kLocalizedInvalidURLGiven)
-            return false
+            decisionHandler(.cancel)
+            return
         }
+
         // parse project name
         let projectName: String = urlComp.queryItems?
             .first(where: { $0.name == "fname" })?.value?
             .replacingOccurrences(of: "+", with: " ") ?? ""
         if projectName.isEmpty {
             Util.alert(withText: kLocalizedInvalidURLGiven)
-            return false
+            decisionHandler(.cancel)
+            return
         }
+
         // check if project exists
         if Project.projectExists(withProjectName: projectName,
                                  projectID: projectID) {
             Util.alert(withText: kLocalizedProjectAlreadyDownloadedDescription)
-            return false
+            decisionHandler(.cancel)
+            return
         }
 
         let url = URL(string: urlWithoutParams ?? "")
         if let fileManager = CBFileManager.shared() {
             fileManager.delegate = self
             fileManager.downloadProject(from: url, withProjectID: projectID, andName: projectName)
+            setProgress(0.2)
             loadingView?.show()
         }
-        return false
+
+        decisionHandler(.cancel)
+        return
     }
 
     // MARK: - WebView Navigation
 
     @objc func goBack(_ sender: Any?) {
-        if webView.canGoBack {
-            webView.goBack()
+        if webView?.canGoBack ?? false {
```

Comment:
Please compare optional booleans with `webView?.canGoBack == true`. This is more readable than assigning it the opposite default value.